internal/util/metrics/metrics.go

```diff
@@ -0,0 +1,98 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+type Server struct {
+	pipelinesGauge          *prometheus.GaugeVec
+	pipelineCounterMap      map[string]struct{}
+	isbsvcGauge             *prometheus.GaugeVec
+	isbsvcCounterMap        map[string]struct{}
+	numaflowControllerGauge *prometheus.GaugeVec
+}
+
+var (
+	// the pipelineMap contains the information of all running pipeline with key as "name_namespace"
+	pipelinesGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "numaflow_pipelines_running",
+			Help: "Number of Numaflow pipelines running",
+		},
+		[]string{},
+	)
+	isbsvcGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "numaflow_isb_service_running",
+			Help: "Number of Numaflow ISB Service running",
+		},
+		[]string{},
+	)
+	numaflowControllerGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "numaflow_controller_running",
+			Help: "Number of Numaflow controller running",
+		},
+		[]string{"name", "namespace", "version"},
+	)
+)
+
+func NewMetricsServer() *Server {
```

Comment:
Is it possible to reuse the metrics server from controller-runtime?

Reply:
I'm not creating a new metrics server here, here it's registering custom metrics to the existing server, maybe fn name is creating confusion here, i'll rename it.